addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -12,7 +12,12 @@ class ResourceCalendarLeaves(models.Model):
 
     timesheet_ids = fields.One2many('account.analytic.line', 'global_leave_id', string="Analytic Lines")
 
-    def _work_time_per_day(self):
+    def _get_resource_calendars(self):
+        if any(not leave.calendar_id for leave in self):
+            return self.env['resource.calendar'].search([('company_id', 'in', self.company_id.ids)])
+        return self.calendar_id
+
+    def _work_time_per_day(self, resource_calendars):
```

Comment:
```suggestion
    def _work_time_per_day(self, resource_calendars=False):
        ...
        resource_calendars = resource_calendars or self._get_resource_calendars()
```
Isn't that a little softer? :)

Reply:
I saw you had a discussion about that, but I am not sure to understand why we would have a traceback with this suggestion. 🤔

Reply:
If we don't pass resource_calendars then it'll not get any data in `calendars_dict` and try to get calendar data from result gives traceback :slightly_smiling_face:

Reply:
Okay but in the case we don't pass any resource_calendars, we could call the `_get_resource_calendars` method as fallback as Audric suggested, no? 🤔 🙂 

What do you think? 🙂